SCAutolib/models/user.py

```diff
@@ -0,0 +1,153 @@
+                       "pairs before adding a new one.")
+        self._key = key
+
+    @key.deleter
+    def key(self):
+        logger.info("Deleting the current user key.")
+        self._key = None
+
+    @property
+    def cert(self):
+        return self._cert
+
+    @cert.setter
+    def cert(self, cert: Path):
+        logger.warning("Make sure to remove the existing key/cert "
+                       "pairs before adding a new one.")
+        self._cert = cert
+
+    @cert.deleter
+    def cert(self):
+        logger.info("Deleting the current user cert.")
+        self._cert = None
+
+    @property
+    def cnf(self):
+        return self._cnf
+
+    @cnf.setter
+    def cnf(self, cnf: Path):
+        if self._cnf:
+            logger.warning("Overriding current CNF file.")
+        self._cnf = cnf
+
+    @cnf.deleter
+    def cnf(self):
+        logger.info("Removing current CNF file.")
+        self._cnf = None
+
+    def delete_user(self):
+        logger.info(f"Deleting the user {self.username}")
+        run(['userdel', '-f', self.username], check=True)
+
+    def add_user(self):
+        run(['useradd', '-m', '-p', self.password, self.username], check=True)
+        logger.info(f"Creating new user {self.username}")
+
+
+class IPAUser(User):
+    """
+    This class is used to represent an IPA user.
+    """
+
+    def __init__(self, ipa_server: str, username: str, password: str,
+                 pin: str, card = None, cnf: Path = None, key: Path = None,
+                 cert: Path = None):
+
+        """
+        :param ipa_server: IPA server hostname where the user will be added.
+        :type str
+        :param username: Username for the system user
+        :type: str
+        :param password: Password for the system user
+        :type str
+        :param pin: Smart card pin for the system user
+        :type str
+        :param card: Card to be associated with the user
+        :type Card
+        :param cnf: CNF file to be associated with the user
+        :type Path
+        :param key: Key to be assoicated with the user
+        :type Path
+        :param cert: Certificate to be associated with the user.
+        :type Path
+        """
+
+        super().__init__(username, password, pin, card, cnf, key, cert)
+        self._ipa_server = ipa_server
+        self._meta_client: ClientMeta = ClientMeta(self._ipa_server,
+                                                   verify_ssl=False)
+
+    def add_user(self):
+        r = self._meta_client.user_add(self.username, self.username,
+                                       self.username, self.username,
+                                       o_userpassword=self.password)
+        logger.debug(r)
+        logger.info(f"User {self.username} is added to the IPA server")
+
+    def delete_user(self):
+        r = self._meta_client.user_del(self.username)["result"]
+        logger.debug(r)
+        logger.info(f"User {self.username} is removed from the IPA server")
```

Comment:
Not sure if these methods are needed because this functionality is already implemented in [IPAServerCA](https://github.com/x00Pavel/SCAutolib/blob/f80ce4f9139ded51f590498651e6a1ca2a138d4b/SCAutolib/models/CA.py#L454)

Reply:
Resolved